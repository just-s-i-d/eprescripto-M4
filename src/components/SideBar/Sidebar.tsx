import { SetStateAction, useEffect, useState } from "react";
import { NavLink, useLocation } from "react-router-dom";

import { Menu, Switch, theme } from "antd";
import Sider from "antd/es/layout/Sider";

import { DOCTOR_NAV_LINKS, PATIENT_NAV_LINKS } from "@constants/constants";

type SideBarProps = {
  role: string;
  textColor: string;
  setIsDarkTheme: React.Dispatch<SetStateAction<boolean>>;
  collapsed: boolean;
  setCollapsed: React.Dispatch<SetStateAction<boolean>>;
};
const SideBar: React.FC<SideBarProps> = ({
  role,
  textColor,
  setIsDarkTheme,
  collapsed,
  setCollapsed,
}) => {
  const [pathname, setPathname] = useState<string>("prescription");
  const location = useLocation();
  const [collapsedWidth, setCollapsedWidth] = useState<number>(80);
  const sideBarLinks = role === "doctor" ? DOCTOR_NAV_LINKS : PATIENT_NAV_LINKS;
  const {
    token: { colorPrimaryBg },
  } = theme.useToken();

  useEffect(() => {
    const currentPath = location.pathname.split("/")[1];
    setPathname(currentPath);
  }, [location.pathname]);

  return (
    <Sider
      className="h-[92vh] max-desktop:top-[8vh] left-0 max-desktop:sticky max-[991px]:fixed z-10"
      width={250}
      breakpoint="lg"
      onBreakpoint={(broken) => {
        if (broken) {
          setCollapsedWidth(0);
          setCollapsed(true);
        } else {
          setCollapsedWidth(80);
        }
      }}
      collapsible
      collapsed={collapsed}
      collapsedWidth={collapsedWidth}
      trigger={null}
    >
      <Menu
        mode="inline"
        className="h-full text-xl px-2 relative"
        selectedKeys={[pathname]}
        style={{
          color: textColor,
        }}
      >
        {sideBarLinks.map((element) => (
          <Menu.Item
            key={element.key}
            icon={element.icon}
            className="my-6 py-2 flex"
          >
            <NavLink className="text-lg" to={element.key}>
              {element.label}
            </NavLink>
          </Menu.Item>
        ))}
        <Switch
          className={`absolute bottom-6`}
          style={{ backgroundColor: colorPrimaryBg }}
          checkedChildren="Dark"
          unCheckedChildren="Light"
          onChange={(value) => setIsDarkTheme(value)}
        />
      </Menu>
    </Sider>
  );
};
export default SideBar;
