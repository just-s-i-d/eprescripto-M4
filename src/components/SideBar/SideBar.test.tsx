import { render, screen, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import SideBar from "./Sidebar";
import { act } from "react-dom/test-utils";

describe("SideBar Component", () => {
  const setIsDarkThemeMock = jest.fn();
  const setCollapsedMock = jest.fn();
  it("Test to check if the side bar renders correctly for doctor role", () => {
    const role = "doctor";
    const { asFragment } = render(
      <MemoryRouter>
        <SideBar
          textColor="white"
          role={role}
          setIsDarkTheme={setIsDarkThemeMock}
          collapsed={false}
          setCollapsed={setCollapsedMock}
        />
      </MemoryRouter>,
    );
    expect(screen.getByText("Write Prescription")).toBeInTheDocument();
    expect(screen.getByText("Reviews")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("Test to check if the side bar renders correctly for patient role", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SideBar
          textColor="white"
          role="patient"
          setIsDarkTheme={setIsDarkThemeMock}
          collapsed={false}
          setCollapsed={setCollapsedMock}
        />
      </MemoryRouter>,
    );
    expect(screen.getByText("Prescriptions")).toBeInTheDocument();
    expect(screen.getByText("Doctors")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("Test to check of the theme swtich button works correctly", () => {
    render(
      <MemoryRouter>
        <SideBar
          textColor="white"
          role="doctor"
          setIsDarkTheme={setIsDarkThemeMock}
          collapsed={false}
          setCollapsed={setCollapsedMock}
        />
      </MemoryRouter>,
    );
    const switchElement = screen.getByRole("switch");
    expect(switchElement).toBeInTheDocument();
    act(() => {
      fireEvent.click(switchElement);
    });
    expect(setIsDarkThemeMock).toHaveBeenCalled();
  });
});
