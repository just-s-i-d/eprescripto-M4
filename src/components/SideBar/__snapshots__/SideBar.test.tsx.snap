// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Component Test to check if the side bar renders correctly for doctor role 1`] = `
<DocumentFragment>
  <aside
    class="ant-layout-sider ant-layout-sider-dark h-[92vh] max-xl:top-[8vh] left-0 max-xl:sticky max-lg:fixed z-10"
    style="flex: 0 0 250px; max-width: 250px; min-width: 250px; width: 250px;"
  >
    <div
      class="ant-layout-sider-children"
    >
      <ul
        class="ant-menu ant-menu-root ant-menu-inline ant-menu-light h-full text-xl px-2 relative css-dev-only-do-not-override-1adbn6x"
        data-menu-list="true"
        role="menu"
        tabindex="0"
      >
        <li
          class="ant-menu-item my-6 py-2 flex"
          data-menu-id="rc-menu-uuid-test-dashboard"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            aria-label="home"
            class="anticon anticon-home ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="home"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            <a
              class="text-lg"
              href="/dashboard"
            >
              Dashboard
            </a>
          </span>
        </li>
        <li
          class="ant-menu-item my-6 py-2 flex"
          data-menu-id="rc-menu-uuid-test-prescription"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            aria-label="form"
            class="anticon anticon-form ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="form"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M904 512h-56c-4.4 0-8 3.6-8 8v320H184V184h320c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V520c0-4.4-3.6-8-8-8z"
              />
              <path
                d="M355.9 534.9L354 653.8c-.1 8.9 7.1 16.2 16 16.2h.4l118-2.9c2-.1 4-.9 5.4-2.3l415.9-415c3.1-3.1 3.1-8.2 0-11.3L785.4 114.3c-1.6-1.6-3.6-2.3-5.7-2.3s-4.1.8-5.7 2.3l-415.8 415a8.3 8.3 0 00-2.3 5.6zm63.5 23.6L779.7 199l45.2 45.1-360.5 359.7-45.7 1.1.7-46.4z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            <a
              class="text-lg"
              href="/prescription"
            >
              Write Prescription
            </a>
          </span>
        </li>
        <li
          class="ant-menu-item my-6 py-2 flex"
          data-menu-id="rc-menu-uuid-test-appointments"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            aria-label="schedule"
            class="anticon anticon-schedule ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="schedule"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M928 224H768v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56H548v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56H328v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56H96c-17.7 0-32 14.3-32 32v576c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V256c0-17.7-14.3-32-32-32zm-40 568H136V296h120v56c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-56h148v56c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-56h148v56c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-56h120v496zM416 496H232c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zm0 136H232c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zm308.2-177.4L620.6 598.3l-52.8-73.1c-3-4.2-7.8-6.6-12.9-6.6H500c-6.5 0-10.3 7.4-6.5 12.7l114.1 158.2a15.9 15.9 0 0025.8 0l165-228.7c3.8-5.3 0-12.7-6.5-12.7H737c-5-.1-9.8 2.4-12.8 6.5z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            <a
              class="text-lg"
              href="/appointments"
            >
              Appointments
            </a>
          </span>
        </li>
        <li
          class="ant-menu-item my-6 py-2 flex"
          data-menu-id="rc-menu-uuid-test-reviews"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            aria-label="star"
            class="anticon anticon-star ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="star"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            <a
              class="text-lg"
              href="/reviews"
            >
              Reviews
            </a>
          </span>
        </li>
        <li
          class="ant-menu-item my-6 py-2 flex"
          data-menu-id="rc-menu-uuid-test-profile"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            aria-label="profile"
            class="anticon anticon-profile ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="profile"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656zM492 400h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H492c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zm0 144h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H492c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zm0 144h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H492c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zM340 368a40 40 0 1080 0 40 40 0 10-80 0zm0 144a40 40 0 1080 0 40 40 0 10-80 0zm0 144a40 40 0 1080 0 40 40 0 10-80 0z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            <a
              class="text-lg"
              href="/profile"
            >
              Profile
            </a>
          </span>
        </li>
        <button
          aria-checked="false"
          class="ant-switch absolute bottom-6 css-dev-only-do-not-override-1adbn6x"
          eventkey="tmp_key-5"
          role="switch"
          style="background-color: rgb(230, 244, 255);"
          type="button"
        >
          <div
            class="ant-switch-handle"
          />
          <span
            class="ant-switch-inner"
          >
            <span
              class="ant-switch-inner-checked"
            >
              Dark
            </span>
            <span
              class="ant-switch-inner-unchecked"
            >
              Light
            </span>
          </span>
        </button>
      </ul>
      <div
        aria-hidden="true"
        style="display: none;"
      >
        <button
          aria-checked="false"
          class="ant-switch absolute bottom-6 css-dev-only-do-not-override-1adbn6x"
          eventkey="tmp_key-5"
          role="switch"
          style="background-color: rgb(230, 244, 255);"
          type="button"
        >
          <div
            class="ant-switch-handle"
          />
          <span
            class="ant-switch-inner"
          >
            <span
              class="ant-switch-inner-checked"
            >
              Dark
            </span>
            <span
              class="ant-switch-inner-unchecked"
            >
              Light
            </span>
          </span>
        </button>
      </div>
    </div>
  </aside>
</DocumentFragment>
`;

exports[`SideBar Component Test to check if the side bar renders correctly for patient role 1`] = `
<DocumentFragment>
  <aside
    class="ant-layout-sider ant-layout-sider-dark h-[92vh] max-xl:top-[8vh] left-0 max-xl:sticky max-lg:fixed z-10"
    style="flex: 0 0 250px; max-width: 250px; min-width: 250px; width: 250px;"
  >
    <div
      class="ant-layout-sider-children"
    >
      <ul
        class="ant-menu ant-menu-root ant-menu-inline ant-menu-light h-full text-xl px-2 relative css-dev-only-do-not-override-1adbn6x"
        data-menu-list="true"
        role="menu"
        tabindex="0"
      >
        <li
          class="ant-menu-item my-6 py-2 flex"
          data-menu-id="rc-menu-uuid-test-dashboard"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            aria-label="home"
            class="anticon anticon-home ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="home"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            <a
              class="text-lg"
              href="/dashboard"
            >
              Dashboard
            </a>
          </span>
        </li>
        <li
          class="ant-menu-item my-6 py-2 flex"
          data-menu-id="rc-menu-uuid-test-prescriptions"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            aria-label="form"
            class="anticon anticon-form ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="form"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M904 512h-56c-4.4 0-8 3.6-8 8v320H184V184h320c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V520c0-4.4-3.6-8-8-8z"
              />
              <path
                d="M355.9 534.9L354 653.8c-.1 8.9 7.1 16.2 16 16.2h.4l118-2.9c2-.1 4-.9 5.4-2.3l415.9-415c3.1-3.1 3.1-8.2 0-11.3L785.4 114.3c-1.6-1.6-3.6-2.3-5.7-2.3s-4.1.8-5.7 2.3l-415.8 415a8.3 8.3 0 00-2.3 5.6zm63.5 23.6L779.7 199l45.2 45.1-360.5 359.7-45.7 1.1.7-46.4z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            <a
              class="text-lg"
              href="/prescriptions"
            >
              Prescriptions
            </a>
          </span>
        </li>
        <li
          class="ant-menu-item my-6 py-2 flex"
          data-menu-id="rc-menu-uuid-test-appointments"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            aria-label="schedule"
            class="anticon anticon-schedule ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="schedule"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M928 224H768v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56H548v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56H328v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56H96c-17.7 0-32 14.3-32 32v576c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V256c0-17.7-14.3-32-32-32zm-40 568H136V296h120v56c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-56h148v56c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-56h148v56c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-56h120v496zM416 496H232c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zm0 136H232c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zm308.2-177.4L620.6 598.3l-52.8-73.1c-3-4.2-7.8-6.6-12.9-6.6H500c-6.5 0-10.3 7.4-6.5 12.7l114.1 158.2a15.9 15.9 0 0025.8 0l165-228.7c3.8-5.3 0-12.7-6.5-12.7H737c-5-.1-9.8 2.4-12.8 6.5z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            <a
              class="text-lg"
              href="/appointments"
            >
              Appointments
            </a>
          </span>
        </li>
        <li
          class="ant-menu-item my-6 py-2 flex"
          data-menu-id="rc-menu-uuid-test-doctors"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            aria-label="star"
            class="anticon anticon-star ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="star"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            <a
              class="text-lg"
              href="/doctors"
            >
              Doctors
            </a>
          </span>
        </li>
        <li
          class="ant-menu-item my-6 py-2 flex"
          data-menu-id="rc-menu-uuid-test-profile"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            aria-label="profile"
            class="anticon anticon-profile ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="profile"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656zM492 400h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H492c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zm0 144h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H492c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zm0 144h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H492c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zM340 368a40 40 0 1080 0 40 40 0 10-80 0zm0 144a40 40 0 1080 0 40 40 0 10-80 0zm0 144a40 40 0 1080 0 40 40 0 10-80 0z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            <a
              class="text-lg"
              href="/profile"
            >
              Profile
            </a>
          </span>
        </li>
        <button
          aria-checked="false"
          class="ant-switch absolute bottom-6 css-dev-only-do-not-override-1adbn6x"
          eventkey="tmp_key-5"
          role="switch"
          style="background-color: rgb(230, 244, 255);"
          type="button"
        >
          <div
            class="ant-switch-handle"
          />
          <span
            class="ant-switch-inner"
          >
            <span
              class="ant-switch-inner-checked"
            >
              Dark
            </span>
            <span
              class="ant-switch-inner-unchecked"
            >
              Light
            </span>
          </span>
        </button>
      </ul>
      <div
        aria-hidden="true"
        style="display: none;"
      >
        <button
          aria-checked="false"
          class="ant-switch absolute bottom-6 css-dev-only-do-not-override-1adbn6x"
          eventkey="tmp_key-5"
          role="switch"
          style="background-color: rgb(230, 244, 255);"
          type="button"
        >
          <div
            class="ant-switch-handle"
          />
          <span
            class="ant-switch-inner"
          >
            <span
              class="ant-switch-inner-checked"
            >
              Dark
            </span>
            <span
              class="ant-switch-inner-unchecked"
            >
              Light
            </span>
          </span>
        </button>
      </div>
    </div>
  </aside>
</DocumentFragment>
`;
